clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/provider/view/OpenstackLoadBalancerProvider.groovy

```diff
@@ -88,16 +88,24 @@ class OpenstackLoadBalancerProvider implements LoadBalancerProvider<OpenstackLoa
     !data ? Sets.newHashSet() : data.collect(this.&fromCacheData)
   }
 
+  /**
+   * Convert load balancer cache data to a load balancer domain item.
+   * @param cacheData
+   * @return
+     */
   OpenstackLoadBalancer fromCacheData(CacheData cacheData) {
     OpenstackLoadBalancer loadBalancer = objectMapper.convertValue(cacheData.attributes, OpenstackLoadBalancer)
-    Set<LoadBalancerServerGroup> serverGroups = cacheData.relationships[SERVER_GROUPS.ns].collect { key ->
+    Set<LoadBalancerServerGroup> serverGroups = cacheData.relationships[SERVER_GROUPS.ns]?.findResults { key ->
+      LoadBalancerServerGroup loadBalancerServerGroup = null
       ServerGroup serverGroup = clusterProvider.getServerGroup(loadBalancer.account, loadBalancer.region, Keys.parse(key)['serverGroup'])
-      LoadBalancerServerGroup loadBalancerServerGroup = new LoadBalancerServerGroup(name: serverGroup?.name, isDisabled: serverGroup?.isDisabled())
-      loadBalancerServerGroup.instances = serverGroup?.instances?.collect { instance ->
-        new LoadBalancerInstance(id: ((OpenstackInstance) instance).instanceId, health: instance.health && instance.health.size() > 0 ? instance.health[0] : null)
-      }?.toSet()
+      if (serverGroup) {
+        loadBalancerServerGroup = new LoadBalancerServerGroup(name: serverGroup.name, isDisabled: serverGroup.isDisabled())
+        loadBalancerServerGroup.instances = serverGroup.instances?.collect { instance ->
+          new LoadBalancerInstance(id: ((OpenstackInstance) instance).instanceId, health: instance.health && instance.health.size() > 0 ? instance.health[0] : null)
```

Comment:
I'm pretty sure the `health` object here should reflect the health assigned to the instance by the load balancer, not what's the first health in the instance. This is different from Kubernetes, since Kubernetes has instances manage their own health state.